Arrowgene.Ddon.GameServer/Characters/JobManager.cs

```diff
@@ -509,38 +518,19 @@ public void RemoveAbility(IDatabase database, CharacterCommon character, byte sl
         {
             // TODO: Performance
             List<Ability> equippedAbilities = character.EquippedAbilitiesDictionary[character.Job];
-            lock(equippedAbilities)
+            lock (equippedAbilities)
             {
-                byte removedAbilitySlotNo = Byte.MaxValue;
-                for(int i=0; i<equippedAbilities.Count; i++)
-                {
-                    Ability equippedAbility = equippedAbilities[i];
-                    byte equippedAbilitySlotNo = (byte)(i+1);
-                    if(character.Job == character.Job && equippedAbilitySlotNo == slotNo)
-                    {
-                        equippedAbilities[i] = null;
-                        removedAbilitySlotNo = equippedAbilitySlotNo;
-                        break;
-                    }
-                }
-
-                for(int i=0; i<equippedAbilities.Count; i++)
-                {
-                    Ability equippedAbility = equippedAbilities[i];
-                    byte equippedAbilitySlotNo = (byte)(i+1);
-                    if(character.Job == character.Job)
-                    {
-                        if(equippedAbilitySlotNo > removedAbilitySlotNo)
-                        {
-                            equippedAbilitySlotNo--;
-                        }
-                    }
-                }
+                equippedAbilities[slotNo - 1] = null; // Erase the item from the list
```

Comment:
Abilities after the removed one have to be shifted. This change wont work fine and will lead to inconsistent state in the client

Reply:
Attempted to address this concern. I want to run some more experiments with the real client. As far as I can tell, everything is working as expected at the vocation NPC. I notice though when going to the augment menu by pressing start/escape it shows out of date augments. When I try to requip the augments and halt in the debugger, the list is filled with all null pointers. Also, the database shows no skills in it. I am not sure how this happens or if there is another packet we need to send to tell the client to update this information.